packages/plugins/tsconfig.json

```diff
@@ -0,0 +1,17 @@
+{
+	"extends": "../../tsconfig.base.json",
+	"compilerOptions": {
+		"rootDir": "src",
+		"declarationDir": "build-types",
+		"types": [ "gutenberg-env" ],
+		"checkJs": false
```

Comment:
Well, I agree it's definitely not ideal. But I think there are two parts of the project:

1. Let typescript infer everything it can from JSDoc, and publish those types for consumers.
2. Migrate to typescript (or at least turn on full type checking), solving every type issue along the way.

My ultimate goal is number 1. Number 2 would be amazing too, but it can _really_ severely impact the amount of work required. In something like data-controls, where there's only a single file, it's very easy.

But the real problem is that some of these packages are absolutely massive, and have thousands of type errors. block-editor is a better example where that would be impossible to do as part of this project.

I don't think we can increase the scope of this work to complete type safety in every one of these projects. My main goal is to get published types to replace DT asap, and I think that can be done without fixing every TS error in these existing projects. 

I also don't think it makes sense to block publishing build types on having completely correct internal types. The DT types don't enforce that, for example, and they are already broken for many packages. So publishing the build-types should result in a much better situation, even before every internal type issue is resolved.

Reply:
Plugins isn't the best example of all this, so I can take another look and see if it wouldn't bee too much work to turn this on.

But you raise a good point and I wanted to discuss it in the context of the broader project, especially since the remaining packages like editor and block-editor are significantly larger

Reply:
(rich-text as well for that matter)